test/test_puddleworldOntology.py

```diff
@@ -33,167 +33,167 @@
+   r"in_half(find(model,\x.heart(x)),left)",
    True),
 
   ("test is_edge",
    SIMPLE_SCENE,
-   r"is_edge(unique(\x.circle(x)))",
+   r"is_edge(find(model,\x.circle(x)))",
    True),
 
   ("test is_edge",
    SIMPLE_SCENE,
-   r"is_edge(unique(\x.star(x)))",
+   r"is_edge(find(model,\x.star(x)))",
    True),
 
   ("test is_edge",
    SIMPLE_SCENE,
-   r"is_edge(unique(\x.diamond(x)))",
+   r"is_edge(find(model,\x.diamond(x)))",
    True),
 
   ("test is_edge",
    SIMPLE_SCENE,
-   r"is_edge(unique(\x.heart(x)))",
+   r"is_edge(find(model,\x.heart(x)))",
    False),
 
   ("test complex query",
    SIMPLE_SCENE,
-   r"horse(unique(\x.relate(x,unique(\y.diamond(y)),down)))",
+   r"horse(find(model,\x.relate(x,find(model,\y.diamond(y)),down)))",
    True),
 
   ("test complex query",
    SIMPLE_SCENE,
-   r"horse(unique(\x.and_(horse(x),relate(x,unique(\y.diamond(y)),down))))",
+   r"horse(find(model,\x.and_(horse(x),relate(x,find(model,\y.diamond(y)),down))))",
    True),
 
 ]
 
 EC_EXPRESSIONS = [
-  "(lambda (ec_unique $0 spade))", # test basic object predicate
-  "(lambda (move (ec_unique $0 spade)))", # test pick
-  "(lambda (relate (ec_unique $0 spade) (ec_unique $0 puddle) down))", # test relate down
-  "(lambda (relate (ec_unique $0 puddle) (ec_unique $0 spade) down))", # test relate down
-  "(lambda (relate (ec_unique $0 spade) (ec_unique $0 puddle) up))", # test relate up
-  "(lambda (relate (ec_unique $0 puddle) (ec_unique $0 spade) up))", # test relate up
-  "(lambda (relate (ec_unique $0 rock) (ec_unique $0 star) left))", # test relate left
-  "(lambda (relate (ec_unique $0 rock) (ec_unique $0 puddle) left))", # test relate left
-  "(lambda (relate (ec_unique $0 rock) (ec_unique $0 star) right))", # test relate right
-  "(lambda (relate (ec_unique $0 star) (ec_unique $0 rock) right))", # test relate right
-  "(lambda (relate (ec_unique $0 puddle) (ec_unique $0 rock) right))", # test relate right
-  "(lambda (relate_n (ec_unique $0 star) (ec_unique $0 rock) right 1))", # test relate_n 1
-  "(lambda (relate_n (ec_unique $0 star) (ec_unique $0 spade) up 2))", # test relate_n 2,
-  "(lambda (in_half (ec_unique $0 star) up))", # test in half
-  "(lambda (in_half (ec_unique $0 star) right))", # test in half
-  "(lambda (in_half (ec_unique $0 star) down))", # test in half
-  "(lambda (in_half (ec_unique $0 star) left))", # test in half
-  "(lambda (in_half (ec_unique $0 triangle) up))", # test in half
-  "(lambda (in_half (ec_unique $0 triangle) right))", # test in half
-  "(lambda (in_half (ec_unique $0 triangle) down))", # test in half
-  "(lambda (in_half (ec_unique $0 triangle) left))", # test in half
-  "(lambda (is_edge (ec_unique $0 circle)))", # test is_edge
-  "(lambda (is_edge (ec_unique $0 star)))", # test is_edge
-  "(lambda (is_edge (ec_unique $0 rock)))", # test is_edge
-  "(lambda (is_edge (ec_unique $0 heart)))", # test is_edge
+  "(lambda (find $0 spade))", # test basic object predicate
```

Comment:
@CatherineWong Any strong reason to write it this way, rather than using a constant `scene`? I guess it allows you to be context-sensitive about the domain of reference?

Reply:
Maybe I'm missing something: what would be the alternative you're suggesting? To evaluate on different scenes, EC must be passed them as arguments (single input arguments) to the discovered program.